### PR TITLE
Add support for AWS CodePipeline

### PR DESCRIPTION
**Pull request by albertvaka**
Add code to extract env vars & spec test.

### CODE REVIEW

File: internal/utils/ci_providers.go [hunk 1]
```diff
@@ -20,19 +20,20 @@ import (
...
+	"BUILDKITE":           extractBuildkite,
+	"CIRCLECI":            extractCircleCI,
+	"GITHUB_SHA":          extractGithubActions,
+	"GITLAB_CI":           extractGitlab,
+	"JENKINS_URL":         extractJenkins,
+	"TEAMCITY_VERSION":    extractTeamcity,
+	"TRAVIS":              extractTravis,
+	"BITRISE_BUILD_SLUG":  extractBitrise,
+	"CF_BUILD_ID":         extractCodefresh,
+	"CODEBUILD_INITIATOR": extractAwsCodePipelines,
```
Comment (tonyredondo) [inline internal/utils/ci_providers.go:36]: you need to compare the value of `CODEBUILD_INITIATOR` with StartWith `codepipeline`

Comment (albertvaka) [reply internal/utils/ci_providers.go:36]: I think this should be enough, and it simplifies the code. Wdyt @andrea-mosk ?

Comment (andrea-mosk) [reply internal/utils/ci_providers.go:36]: Yeah, unfortunately CodeBuild builds can be started by a variety of other AWS services. This won't be present only for CodePipeline, but also for all the other services, so we need to make sure it starts with `codepipeline`

Comment (albertvaka) [reply internal/utils/ci_providers.go]: Should be fixed now :)

**Review by tonyredondo — APPROVED** [review]

**Pull request merged**